agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -16,26 +17,25 @@ class ServicePodManager
 
     def initialize(autostart = true)
       @workers = {}
-      subscribe('agent:node_info', :on_node_info)
-      subscribe('service_pod:update', :on_update_notify)
-      async.start if autostart
+
+      if autostart
+        observe(Actor[:node_info_worker]) do |node|
```

Comment:
Should this observe be inside `#start` .. how can actor start observing if `autostart=false` and start is called manually?